### ISSUE TITLE
[3.x] Repeatedly calling JavaScript->Godot callback causes infinite memory consumption increases.

### ISSUE THREAD

**Issue by SysError99** [godotengine/godot#81005]
### Godot version

3.5.2-stable

### System information

Linux 6.1.44-1-MANJARO

### Issue description

Calling Godot Callback from JavaScript causes memory consumption to go higher than expected, even with empty project that does nothing but keep calling Godot's callback via JavaScript or Godot's JavaScript interface.

### Steps to reproduce

1. On the game's script. create Godot Callback for JavaScript.
2. Bind the callback to reachable JavaScript interface (`window` in particular)
3. Call the callback by either using native JavaScript code on browser, or from Godot's JavaScript interface in high amount of times per second (in the reproduce, ~300 times per second).
4. Memory consumption in JavaScript's heap keeps increasing.

This happens on both Firefox and Chromium-based browsers, WebKit is untested.
![image](https://github.com/godotengine/godot/assets/17522480/d82c9009-1433-4769-9741-b726e15cbd37)
![image](https://github.com/godotengine/godot/assets/17522480/7f7bf2de-2aad-472c-bf80-ac3165667c43)

### Minimal reproduction project

[Archive.zip](https://github.com/godotengine/godot/files/12443735/Archive.zip)

**Comment by SysError99** [comment]
I left it for awhile and it still keeps increasing, on an empty project.
![image](https://github.com/godotengine/godot/assets/17522480/f9baf985-4a03-4ce4-996b-cee7d31244e3)

**Comment by SysError99** [comment]
Further testing on bleeding edge `3.x` branch found out that after disabling internal call line from `JavaScriptObjectImpl::_callback`, its memory footprint grows much less, however it's still growing. Suggesting that something is wrong in some sort of memory reserving mechanism in this function.

![image](https://github.com/godotengine/godot/assets/17522480/bcb98dbf-125a-40e2-ad86-3b193986f762)
![image](https://github.com/godotengine/godot/assets/17522480/ceb083f8-b957-48ae-bee4-370352c4621c)

Since my experience with C++ and manual memory management is still very limited, I cannot analyse the problem any further, but I will try my test to figure out the culprit.

**Comment by SysError99** [comment]
Looks like the problem sits down somewhere at the callback function, since repeatedly calling it even with zero arguments still pile the array up. GodotJsWrapper seemed to not free the arg passing array on callback for some reason.

![Clipboard - August 28, 2023 7_25 PM](https://github.com/godotengine/godot/assets/17522480/c12b2a7e-8121-4c2a-8bbc-09ff3f4b8c9f)

**Comment by SysError99** [comment]
As for heads up, I found a culprit that `GodotJsWrapper` never frees up freshy-created and 'proxied' array in the callback. I wrapped the array with `GodotJSWrapper.MyProxy` instead of `GodotJSWrapper.get_proixed` and then free out the proxy manually after callback execution.

![image](https://github.com/godotengine/godot/assets/17522480/03d97838-b441-4059-934a-0b0799895fc3)

Still, I can't be too sure if this is the 'right' solution since JS GC can be wonky to deal with especially when interfacing with WASM. However, as for my test bench code it doesn't ruin anything and value still gets copied and referenced properly.

*This message is mirrored from Godot Contributors Chat.*
